core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -226,7 +226,11 @@ object SparkSubmit {
 
     // Set the cluster manager
     val clusterManager: Int = args.master match {
-      case m if m.startsWith("yarn") => YARN
+      case "yarn" => YARN
+      case "yarn-client" | "yarn-cluster" =>
```

Comment:
if I specify `yarn-anything` now this will fail with an error message that is not helpful:
`Master must start with yarn, spark, mesos, or local` (my master *does* start with yarn). We should handle that case or change the error message.